src/perfctr.c

```diff
@@ -151,7 +151,7 @@ static void destroy_shared_region(void)
 
 	if (shmem_fd > 0)
 		shm_unlink(shmem_name);
-	if (sem && sem != SEM_FAILED) {
+	if (sem != SEM_FAILED) {
```

Comment:
This implies an assumption that SEM_FAILED == 0. Maybe we should also explicitly initialize sem with SEM_FAILED, and reset it to SEM_FAILED instead of NULL (we have it in the error path in init_shared_region, but missing it here anyway).